src/mpi/comm/comm_split_type.c

```diff
@@ -73,6 +70,142 @@ int MPIR_Comm_split_type_node(MPIR_Comm * comm_ptr, int key, MPIR_Comm ** newcom
     goto fn_exit;
 }
 
+#ifdef HAVE_HWLOC
+
+struct shmem_info_table {
+    const char *val;
+    hwloc_obj_type_t hwloc_obj_type;
+    int depth;
+};
+
+static struct shmem_info_table shmem_info[] = {
+    {"socket", HWLOC_OBJ_PACKAGE, -1},
+    {"package", HWLOC_OBJ_PACKAGE, -1},
+    {"numanode", HWLOC_OBJ_NUMANODE, -1},
+    {"core", HWLOC_OBJ_CORE, -1},
+    {"hwthread", HWLOC_OBJ_PU, -1},
```

Comment:
There should be an alias for "pu" to point to `HWLOC_OBJ_PU` as well.

Reply:
We should probably add equivalents for other objects in hwloc too, e.g., `MACHINE`.